### DISCUSSION TITLE
E2E Test for provision recipe

### DISCUSSION THREAD

**Discussion by bsacharski** [deployphp/deployer#2227] [General]
I've looked into running E2E test for `provision` recipe in docker environment - while installing packages, setting users and so on is fine, the process fails during the `provision:firewall` step.

The `ufw` is, from what I can tell, a sort of a nicer frontend to `iptables`. `iptables` is a kernel module, so unfortunately it does not work nicely with containerized environments that share kernel with the host OS.

I think that at this point, it looks like for provision recipe E2E test to work, it would have to be executed against an environment with its own separate kernel, like a VM. At the same time I have no idea whether the GitHub Actions can spawn a dedicated VM with SSH access for free though.

**Comment by antonmedv** [comment]
I think we can make an option to disable configuring iptables, and test with `dep provision -o firewall=false`.

**Comment by bsacharski** [comment]
Ok, I will proceed with that then
